### PR TITLE
Add a sniff to avoid long functions

### PR DESCRIPTION
**Pull request by 50bhan**
Following issue #1099.

### CODE REVIEW

**Comment by 50bhan** [comment]
@kukulich Hi, I believe this PR is ready for review. I'd appreciate it if you could check it out. 

P.S. I'm going to prepare seven PRs for new sniffs. I sent four of them already and I need this PR helper for remaining PRs. That's the reason it's more convenient to go with this one first.

File: SlevomatCodingStandard/Helpers/StructureMetricHelper.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php declare(strict_types = 1);
+
+namespace SlevomatCodingStandard\Helpers;
+
+use PHP_CodeSniffer\Files\File;
+
+class StructureMetricHelper
+{
+
+	public static function getFunctionLengthInLines(File $file, int $position): int
```
Comment (kukulich) [inline SlevomatCodingStandard/Helpers/StructureMetricHelper.php:10]: Please move method to `FunctionHelper`.

File: SlevomatCodingStandard/Helpers/StructureMetricHelper.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	{
+		$tokens = $file->getTokens();
+		$token = $tokens[$position];
+
+		// Skip function without body.
+		if (!isset($token['scope_opener'])) {
```
Comment (kukulich) [inline SlevomatCodingStandard/Helpers/StructureMetricHelper.php:16]: Use `FunctionHelper::isAbstract()`

File: SlevomatCodingStandard/Sniffs/Files/FunctionLengthSniff.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+use PHP_CodeSniffer\Sniffs\Sniff;
+use SlevomatCodingStandard\Helpers\StructureMetricHelper;
+use function sprintf;
+use const T_FUNCTION;
+
+final class FunctionLengthSniff implements Sniff
+{
+
+	public const CODE_FUNCTION_LENGTH = 'FunctionLength';
+	private const ERROR_MESSAGE = 'Your function is too long. Currently using %d lines. Can be up to %d lines.';
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Files/FunctionLengthSniff.php:15]: Useless constant.

File: SlevomatCodingStandard/Sniffs/Files/FunctionLengthSniff.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+	/** @var int */
+	public $maxLength = 20;
+
+	/**
+	 * @return int[]
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Files/FunctionLengthSniff.php:21]: Please use `array<int, (int|string)>`

File: SlevomatCodingStandard/Sniffs/Files/FunctionLengthSniff.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	 */
+	public function register(): array
+	{
+		return [T_FUNCTION];
+	}
+
+	/**
+	 * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
+	 * @param File $file
+	 * @param int $position
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Files/FunctionLengthSniff.php:31]: `$functionPointer`

File: SlevomatCodingStandard/Sniffs/Files/FunctionLengthSniff.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+	 */
+	public function process(File $file, $position): void
+	{
+		$length = StructureMetricHelper::getFunctionLengthInLines($file, $position);
+
+		if ($length <= $this->maxLength) {
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Files/FunctionLengthSniff.php:37]: Use `SniffSettingsHelper::normalizeInteger()`

File: README.md [hunk 7]
```diff
@@ -242,6 +242,12 @@ Some functions have `$strict` parameter. This sniff reports calls to these funct
 
 Reports closures not using `$this` that are not declared `static`.
 
+#### SlevomatCodingStandard.Functions.FunctionLength 🔧
+
+Disallows long functions. This sniff provides the following setting:
+
+* `maxLength`: specifies max allowed function length (default value is 20).
```
Comment (kukulich) [inline README.md:249]: Please `maxLinesLength`

**Review by kukulich — CHANGES_REQUESTED** [review]
Please fix code coverage.

**Pull request merged**